File: BackendSrc/Domain/TimetableEntities/GuidebookEntities/LessonTime.cs
using Domain.TimetableEntities.TeacherEntities;

namespace Domain.TimetableEntities.GuidebookEntities
{
    public class LessonTime : Entity
    {
        public int Year { get; set; }
        public TimeOnly StartTime { get; set; }
        public TimeOnly EndTime { get; set; }

        public List<TeacherAvailableHours> TeacherAvailableHours { get; } = new List<TeacherAvailableHours>();

        public LessonTime( int year, TimeOnly startTime, TimeOnly endTime )
        {
            Year = year;
            StartTime = startTime;
            EndTime = endTime;
        }
    }
}

File: Frontend/src/core/Schedule/LessonTime.ts
import { Time } from "./Time";

export class LessonTime {
  public startTime: Time;
  public endTime: Time;

  constructor(_startTime: Time, _endTime: Time) {
    this.startTime = _startTime;
    this.endTime = _endTime;
  }

  public equalsTo(lessonTime: LessonTime): boolean {
    return this.startTime.equalsTo(lessonTime.startTime) && this.endTime.equalsTo(lessonTime.endTime);
  }

  public toString(): string {
    return this.startTime.toString() + "-" + this.endTime.toString();
  }
}
